File: static/csharp/ModelOps.cs
using System.Threading;
using System.Threading.Tasks;

namespace MyAirtable;

/// <summary>
/// Static helpers backing the generated per-model fluent CRUD methods. C# has reified generics, so
/// these need no class token (unlike the Java target): the generated <c>{Table}Model</c> emits
/// covariant <c>SaveAsync</c>/<c>FetchAsync</c>/<c>DeleteAsync</c> methods that delegate here.
/// </summary>
public static class ModelOps
{
    /// <summary>
    /// Persist the model's dirty fields back to Airtable. Requires a saved record (<c>Id != null</c>)
    /// — construct a fresh model and pass it to the table's <c>CreateAsync(...)</c> to insert a
    /// brand-new row.
    /// </summary>
    public static Task<T> SaveAsync<T>(
        T model,
        bool typecast = false,
        CancellationToken ct = default
    )
        where T : AirtableModel
    {
        var client = model.RequireAttachedClient();
        if (model.IsNew)
            throw new AirtableException.ApiError(
                "UNSAVED_MODEL",
                "Cannot save a model without an id; use the table's CreateAsync(...) instead"
            );
        return new OrmTable<T>(model.TableId, client).UpdateAsync(model, typecast, ct);
    }

    /// <summary>Re-fetch the model from the server. Returns a fresh instance.</summary>
    public static Task<T> FetchAsync<T>(T model, CancellationToken ct = default)
        where T : AirtableModel
    {
        var client = model.RequireAttachedClient();
        if (model.IsNew)
            throw new AirtableException.ApiError("UNSAVED_MODEL", "Cannot fetch an unsaved model");
        return new OrmTable<T>(model.TableId, client).GetAsync(model.RequireId(), ct);
    }

    /// <summary>Delete the record referenced by this model's <c>Id</c>.</summary>
    public static Task DeleteAsync(AirtableModel model, CancellationToken ct = default)
    {
        var client = model.RequireAttachedClient();
        if (model.IsNew)
            throw new AirtableException.ApiError("UNSAVED_MODEL", "Cannot delete an unsaved model");
        return client.DeleteRecordAsync(model.TableId, model.RequireId(), ct);
    }
}

File: src/myairtable/static/java/ModelOps.java
// ==========================================
// MyAirtable static runtime.
// ==========================================

package myairtable;

/**
 * Static helpers backing the generated per-model fluent CRUD methods.
 *
 * <p>The other targets resolve the model's table reflectively (Kotlin: {@code inline reified}
 * extensions); Java has no reified generics, so the generator emits covariant {@code save()}/{@code
 * fetch()}/{@code delete()} methods on each model that delegate here with an explicit class token
 * (java-plan-v2 §2.3.12).
 */
public final class ModelOps {

  private ModelOps() {}

  /**
   * Persist the model's dirty fields back to Airtable. Requires a saved record ({@code id != null})
   * — construct a fresh model and pass it to the table's {@code create(...)} to insert a brand-new
   * row.
   */
  public static <T extends AirtableModel> T save(T model, Class<T> type) {
    AirtableClient client = model.requireAttachedClient();
    if (model.isNew()) {
      throw new AirtableException.Api(
          "UNSAVED_MODEL",
          "Cannot save a model without an id; use Airtable.<table>.create(...)" + " instead");
    }
    return new OrmTable<>(model.getTableId(), type, client).update(model);
  }

  /** Re-fetch the model from the server. Returns a fresh instance. */
  public static <T extends AirtableModel> T fetch(T model, Class<T> type) {
    AirtableClient client = model.requireAttachedClient();
    if (model.isNew()) {
      throw new AirtableException.Api("UNSAVED_MODEL", "Cannot fetch an unsaved model");
    }
    return new OrmTable<>(model.getTableId(), type, client).get(model.getId());
  }

  /** Delete the record referenced by this model's {@code id}. */
  public static void delete(AirtableModel model) {
    AirtableClient client = model.requireAttachedClient();
    if (model.isNew()) {
      throw new AirtableException.Api("UNSAVED_MODEL", "Cannot delete an unsaved model");
    }
    client.deleteRecord(model.getTableId(), model.getId());
  }
}
